lib/spack/spack/binary_distribution.py

```diff
@@ -703,10 +703,23 @@ def generate_package_index(cache_prefix):
                            enable_transaction_locking=False,
                            record_fields=['spec', 'ref_count'])
 
-    file_list = (
-        entry
-        for entry in web_util.list_url(cache_prefix)
-        if entry.endswith('.yaml'))
+    try:
+        file_list = (
+            entry
+            for entry in web_util.list_url(cache_prefix)
+            if entry.endswith('.yaml'))
+    except KeyError as inst:
```

Comment:
Is this the same `KeyError` that we talked about from the s3 implementation of `list_url`?  I thought we had addressed this error?

Reply:
It's only addressed here, as far as I know, unless you got a fix merged at some point?